localstack-core/localstack/aws/handlers/validation.py

```diff
@@ -51,7 +56,8 @@ def __call__(self, chain: HandlerChain, context: RequestContext, response: Respo
 
         if path.startswith(f"{INTERNAL_RESOURCE_PATH}/") or path.startswith("/_aws/"):
             try:
-                self.openapi.validate_request(WerkzeugOpenAPIRequest(context.request))
+                for openapi in self.open_apis:
```

Comment:
q: I am curious what is the runtime implication of this is. Since this is executed on every request to the APIs (if enabled), it is time critical. How long does a check with one big "merged" spec take? And how long does it take if you iterate over the specs and perform the validation for each and every spec registered?

I am afraid that the overhead in the handler performing lots of different validations could be slow... Since we currently don't enable the validator by default, it's fine for now, but this could become an issue in the future...

Reply:
That's a fair question. I quickly tried to get some sense of the overhead, and we are talking about ~2 milliseconds to validate a request against the core spec. I suspect that most of the overhead to validate a spec is due to how large the spec is and how deeply nested its objects are. Therefore, I assume (but I can be wrong here) that validating against one large spec would not have a big impact here. In addition, we are only validating "internal" (`_localstack` and `_aws`) requests and we currently have only two specs.